### PR TITLE
[Refactoring] Strong type for IMAP Tag

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: protocols/imap/src/test/java/org/apache/james/imap/decode/parser/GetQuotaRootParserTest.java [hunk 1]
```diff
@@ -42,8 +43,8 @@ public void testNonQuotedMailbox() throws DecodingException {
         String commandString = "INBOX\n";
         InputStream inputStream = new ByteArrayInputStream(commandString.getBytes());
         ImapRequestStreamLineReader lineReader = new ImapRequestStreamLineReader(inputStream, null);
-        GetQuotaRootRequest request = (GetQuotaRootRequest) parser.decode(command, lineReader, "A003", null);
-        GetQuotaRootRequest expected = new GetQuotaRootRequest("A003", command, "INBOX");
+        GetQuotaRootRequest request = (GetQuotaRootRequest) parser.decode(command, lineReader, new Tag("A003"), null);
```
Comment (Arsnael) [inline protocols/imap/src/test/java/org/apache/james/imap/decode/parser/GetQuotaRootParserTest.java]: Why reusing the TAG fixture in other tests files but not this one?

File: protocols/imap/src/test/java/org/apache/james/imap/decode/parser/MoveParserTest.java [hunk 2]
```diff
@@ -41,8 +42,8 @@ public void testQuotaParsing() throws DecodingException {
 
         InputStream inputStream = new ByteArrayInputStream(commandString.getBytes());
         ImapRequestStreamLineReader lineReader = new ImapRequestStreamLineReader(inputStream, null);
-        MoveRequest request = (MoveRequest) parser.decode(command, lineReader, "A003", null);
-        MoveRequest expected = new MoveRequest(command, new IdRange[] {new IdRange(42, 69)}, "foo", false, "A003");
+        MoveRequest request = (MoveRequest) parser.decode(command, lineReader, new Tag("A003"), null);
```
Comment (Arsnael) [inline protocols/imap/src/test/java/org/apache/james/imap/decode/parser/MoveParserTest.java]: Idem

File: protocols/imap/src/test/java/org/apache/james/imap/decode/parser/SearchCommandParserQuotedCharsetTest.java [hunk 3]
```diff
@@ -52,7 +53,7 @@
 
     private static final Charset ASCII = Charset.forName("US-ASCII");
 
-    private static final String TAG = "A1";
+    private static final Tag TAG = new Tag("A1");
```
Comment (Arsnael) [inline protocols/imap/src/test/java/org/apache/james/imap/decode/parser/SearchCommandParserQuotedCharsetTest.java]: Idem

File: protocols/imap/src/test/java/org/apache/james/imap/decode/parser/SetQuotaCommandParserTest.java [hunk 4]
```diff
@@ -43,7 +44,7 @@ public void testQuotaParsing() throws DecodingException {
         String commandString = "quotaRoot (STORAGE 512) ( MESSAGE  1024  ) \n";
         InputStream inputStream = new ByteArrayInputStream(commandString.getBytes());
         ImapRequestStreamLineReader lineReader = new ImapRequestStreamLineReader(inputStream, null);
-        SetQuotaRequest request = (SetQuotaRequest) parser.decode(command, lineReader, "A003", null);
+        SetQuotaRequest request = (SetQuotaRequest) parser.decode(command, lineReader, new Tag("A003"), null);
```
Comment (Arsnael) [inline protocols/imap/src/test/java/org/apache/james/imap/decode/parser/SetQuotaCommandParserTest.java]: Idem?

File: protocols/imap/src/test/java/org/apache/james/imap/decode/parser/StoreCommandParserTest.java [hunk 5]
```diff
@@ -57,7 +58,7 @@ public void testShouldParseSilentDraftFlagged() throws Exception {
         flags.add(Flags.Flag.DRAFT);
         flags.add(Flags.Flag.FLAGGED);
         check("1 FLAGS.SILENT (\\Draft \\Flagged)\r\n", ranges, true, null,
-                flags, false, "A01");
+                flags, false, new Tag("A01"));
```
Comment (Arsnael) [inline protocols/imap/src/test/java/org/apache/james/imap/decode/parser/StoreCommandParserTest.java]: Idem?

File: protocols/imap/src/test/java/org/apache/james/imap/processor/GetQuotaRootProcessorTest.java [hunk 6]
```diff
@@ -90,7 +91,7 @@ public void setUp() {
 
     @Test
     public void processorShouldWorkOnValidRights() throws Exception {
-        GetQuotaRootRequest getQuotaRootRequest = new GetQuotaRootRequest("A004", ImapCommand.anyStateCommand("Name"), "INBOX");
+        GetQuotaRootRequest getQuotaRootRequest = new GetQuotaRootRequest(new Tag("A004"), ImapCommand.anyStateCommand("Name"), "INBOX");
```
Comment (Arsnael) [inline protocols/imap/src/test/java/org/apache/james/imap/processor/GetQuotaRootProcessorTest.java]: Idem?

**Comment by Arsnael** [comment]
```
[ERROR]   RabbitMQAuthenticatedStateTest.testStatusKOREA » Runtime java.lang.IllegalStat...
```
?

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
```
11:17:40.858 [ERROR] o.a.j.t.SerialTaskManagerWorker - Error while running task TaskId{value=a3c1fc30-bc44-4736-ac14-f8bee03f1567}
[bd163a4ac45aed0fea236a2fed1931c578f52dae] java.lang.IllegalArgumentException: Invalid UUID string: 453
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at java.base/java.util.UUID.fromString(UUID.java:215)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at org.apache.james.mailbox.cassandra.ids.CassandraId$Factory.fromString(CassandraId.java:35)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at org.apache.james.mailbox.cassandra.ids.CassandraId$Factory.fromString(CassandraId.java:32)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at org.apache.james.event.json.JsonSerialize.org$apache$james$event$json$JsonSerialize$$$anonfun$mailboxIdReads$1(EventSerializer.scala:238)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at org.apache.james.event.json.JsonSerialize$$anonfun$15.reads(EventSerializer.scala:237)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.PathReads.$anonfun$at$2(JsConstraints.scala:31)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.JsSuccess.flatMap(JsResult.scala:20)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.PathReads.$anonfun$at$1(JsConstraints.scala:31)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.Reads$$anon$6.reads(Reads.scala:195)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.Reads.$anonfun$map$1(Reads.scala:44)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.Reads$$anon$6.reads(Reads.scala:195)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.Reads$$anon$1$$anon$2.reads(Reads.scala:173)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.Reads$$anon$1$$anon$2.reads(Reads.scala:173)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.Reads$$anon$1$$anon$2.reads(Reads.scala:173)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.Reads.$anonfun$map$1(Reads.scala:44)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.Reads$$anon$6.reads(Reads.scala:195)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at julienrf.json.derived.DerivedReadsInstances$$anon$2.$anonfun$reads$2(DerivedReads.scala:39)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.Reads$$anon$6.reads(Reads.scala:195)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.PathReads.$anonfun$at$2(JsConstraints.scala:31)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.JsSuccess.flatMap(JsResult.scala:20)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.PathReads.$anonfun$at$1(JsConstraints.scala:31)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.Reads$$anon$6.reads(Reads.scala:195)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.Reads.$anonfun$map$1(Reads.scala:44)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.Reads$$anon$6.reads(Reads.scala:195)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.Reads$$anon$3$$anon$4.reads(Reads.scala:180)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.Reads.$anonfun$map$1(Reads.scala:44)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.Reads$$anon$6.reads(Reads.scala:195)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.Reads$$anon$3$$anon$4.reads(Reads.scala:182)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.Reads.$anonfun$map$1(Reads.scala:44)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.Reads$$anon$6.reads(Reads.scala:195)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.Reads$$anon$3$$anon$4.reads(Reads.scala:182)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.Reads.$anonfun$map$1(Reads.scala:44)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.Reads$$anon$6.reads(Reads.scala:195)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.Reads$$anon$3$$anon$4.reads(Reads.scala:182)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.Reads.$anonfun$map$1(Reads.scala:44)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.Reads$$anon$6.reads(Reads.scala:195)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.Reads$$anon$3$$anon$4.reads(Reads.scala:182)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.Reads.$anonfun$map$1(Reads.scala:44)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.Reads$$anon$6.reads(Reads.scala:195)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.OFormat$$anon$4.reads(Format.scala:47)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at play.api.libs.json.Json$.fromJson(Json.scala:191)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at org.apache.james.event.json.JsonSerialize$EventSerializerPrivateWrapper.fromJson(EventSerializer.scala:327)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at org.apache.james.event.json.JsonSerialize.fromJson(EventSerializer.scala:332)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at org.apache.james.event.json.EventSerializer.fromJson(EventSerializer.scala:340)
[bd163a4ac45aed0fea236a2fed1931c578f52dae] 	at org.apache.james.mailbox.events.CassandraEventDeadLettersDAO.deserializeEvent(CassandraEventDeadLettersDAO.java:115)

[bd163a4ac45aed0fea236a2fed1931c578f52dae] [INFO] Tests run: 21, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 45.11 s - in org.apache.james.webadmin.integration.WebAdminServerTaskSerializationIntegrationTest
[bd163a4ac45aed0fea236a2fed1931c578f52dae] [INFO] 
[bd163a4ac45aed0fea236a2fed1931c578f52dae] [INFO] Results:
[bd163a4ac45aed0fea236a2fed1931c578f52dae] [INFO] 
[bd163a4ac45aed0fea236a2fed1931c578f52dae] [ERROR] Failures: 
[bd163a4ac45aed0fea236a2fed1931c578f52dae] [ERROR]   EventDeadLettersIntegrationTest.groupIdOfFailedEventShouldBeStoredInDeadLetter:240 1 expectation failed.


[bd163a4ac45aed0fea236a2fed1931c578f52dae] JSON path . doesn't match.
[bd163a4ac45aed0fea236a2fed1931c578f52dae] Expected: iterable with items ["org.apache.james.webadmin.integration.EventDeadLettersIntegrationTest$RetryEventsListenerGroup"] in any order
[bd163a4ac45aed0fea236a2fed1931c578f52dae]   Actual: []
```

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
`WithEncryptedSwiftTest.mailsShouldBeWellReceived » ConditionTimeout Condition ...`

test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**